contrib/src/main/java/com/twelvemonkeys/contrib/tiff/TIFFUtilities.java

```diff
@@ -200,11 +200,10 @@ public static void rotatePage(ImageInputStream imageInput, ImageOutputStream ima
     }
 
     public static List<TIFFPage> getPages(ImageInputStream imageInput) throws IOException {
-        ArrayList<TIFFPage> pages = new ArrayList<TIFFPage>();
-
         CompoundDirectory IFDs = (CompoundDirectory) new TIFFReader().read(imageInput);
 
-        int pageCount = IFDs.directoryCount();
+        final int pageCount = IFDs.directoryCount();
+        List<TIFFPage> pages = new ArrayList<>(pageCount);
```

Comment:
In this particular case, it possible to reliably determine the exact size that will be used.